plugins/thunderbolt/fu-plugin-thunderbolt.c

```diff
@@ -221,8 +280,16 @@ fu_plugin_thunderbolt_add (FuPlugin *plugin, GUdevDevice *device)
 		}
 	}
 	if (!is_safemode) {
+		if (is_host)
+			if (!fu_plugin_thunderbolt_is_native (device, &is_native, &error)) {
+				g_error ("failed to get native mode status: %s", error->message);
```

Comment:
Never g_error() from a plugin; this basically crashes the daemon, which means I get an email thanks to abrt every time someone hits this. It really does really just need logging into the journal with g_warning...

Reply:
Ouch. Will change it back.

Reply:
I think g_error is fine if it's really fatal i.e. the daemon isn't usable at all, for instance if the XML D-Bus schema can't be loaded. In this case you can still use fwupd with all the other plugins, and so it's not really "fatal", just "one for the logs". But yes, ouch.

Reply:
I'm also on vacation until January, but if you fix up this I think we can merge to master. I'm in two minds about doing a Christmas release; in one sense there is a lot of new device support, but in the other case we're all on holidays and not in a position to test things.

Reply:
I plan to have this fixed tomorrow and test it on Sunday, hopefully.